textile/chat-features.textile

```diff
@@ -9,5 +9,652 @@ jump_to:
+*** @(CHA-RL4b2)@ This specification point has been removed.
+*** @(CHA-RL4b3)@ This specification point has been removed.
+*** @(CHA-RL4b4)@ This specification point has been removed.
+*** @(CHA-RL4b5)@ @[Testable]@ If a room lifecycle operation is not in progress and the channel state is @FAILED@, then the room status shall be transitioned to @FAILED@, using the @reason@ for the channel state change as the @error@ for the room status change. All @transient disconnect timeouts@ are cancelled and a @CHA-RL2f@ detach procedure is performed.
+*** @(CHA-RL4b6)@ This specification point has been removed.
+*** @(CHA-RL4b7)@ @[Testable]@ If a room lifecycle operation is not in progress and the channel state is @ATTACHING@ and no transient disconnect timeout exists for the contributor, then a transient disconnect timeout with a 5 second limit is created for the contributor. Upon timeout, the room status is transitioned to @ATTACHING@, using the @reason@ from the initial channel state change as the error for the transition.
+*** @(CHA-RL4b10)@ @[Testable]@ If a room lifecycle operation is not in progress and the channel state is @ATTACHED@ and a transient disconnect timeout exists for the contributor, the timeout is cleared.
+*** @(CHA-RL4b8)@ @[Testable]@ If a room lifecycle operation is not in progress, the channel state is @ATTACHED@, the room status is NOT @ATTACHED@ and all contibutors channel are now @ATTACHED@, the room status is transitioned to @ATTACHED@.
+*** @(CHA-RL4b9)@ @[Testable]@ If a room lifecycle operation is not in progress and the channel state is @SUSPENDED@, then the room status is transitioned to @SUSPENDED@, using the @reason@ of the channel state change as the error. Any transient disconnect timeouts are cancelled and the room enters the @RETRY@ loop.
+
+
+h2(#room-configuration). Room Configuration
+
+Each chat room can be configured individually, allowing options to be passed as to which features are enabled and also feature-specific settings.
+
+* @(CHA-RC1)@ Chat Rooms are singleton objects with respect to the Chat Client on which they are created.
+** @(CHA-RC1a)@ @[Testable]@ Requesting a room from the Chat Client will provide an instance of a room with the provided id.
+** @(CHA-RC1b)@ @[Testable]@ Once a room has been created on the Chat Client, assuming it is not subsequently released, then the client will return the same room instance on subsequent calls for the same room ID.
+** @(CHA-RC1c)@ @[Testable]@ If a room is requested with different options to an existing room of the same id, then an @ErrorInfo@ with code @40000@ is thrown.
+** @(CHA-RC1d)@ @[Testable]@ If a room is requested to be released, it is immediately removed from the SDKs room map - meaning subsequent calls to get an instance of a room for the same ID shall produce a new object.
```

Comment:
Not exactly related to this spec point but there isn’t a more obvious place to put it — what happens if the user calls `rooms.release()` with the ID of a room that doesn't exist in the room map?

Reply:
A good question! I'm in the process of updating this section as part of a recent DR, the changes are here:

https://github.com/ably/specification/pull/217

But in short - it's a no-op :)